### ISSUE TITLE
[receiver/syslog] Syslog Receiver fails to parse long messages, even with a `max_log_size` set

### ISSUE THREAD

**Issue by sinkingpoint** [open-telemetry/opentelemetry-collector-contrib#33182]
### Component(s)

receiver/syslog

## Description

When using the syslog receiver, we can only parse messages up to the default maximum length (8192 octets), even with a `max_log_size` set much higher.

## Steps to Reproduce

1. Create a receiver with the provided config (note the `max_log_size` of 100MiB)
2. Send a message in longer than 8192 characters
3. Observe an error: `message too long to parse. was size 40366, max length 8192`

## Expected Result

The message should parse properly

## Actual Result

The message fails to parse

### Collector version

v0.100.0

## Environment
OS: Debian Bookworm

### OpenTelemetry Collector configuration

```yaml
receivers:
  syslog:
    protocol: rfc5424
    enable_octet_counting: true
    tcp:
      listen_address: :4278
      max_log_size: 100000000 # 100MiB
exporters:
  debug:
service:
  pipelines:
    logs:
      receivers: [syslog]
      exporters: [debug]
```

### Log output

```shell
{"level":"error","ts":1716356887.9432147,"caller":"helper/transformer.go:101","msg":"Failed to process entry","kind":"receiver","name":"syslog/db","data_type":"logs","operator_id":"syslog_input_internal_parser","operator_type ":"syslog_parser","error":"message too long to parse. was size 40366, max length 8192","action":"send","stacktrace":"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*TransformerOperator).HandleEntryError\\n\\tgithub.co m/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.99.0/operator/helper/transformer.go:101\\ngithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*ParserOperator).ParseWith\\n\\tgithub.com/open-telemetry/opentelemetry-collect or-contrib/pkg/stanza@v0.99.0/operator/helper/parser.go:140\\ngithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*ParserOperator).ProcessWithCallback\\n\\tgithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.99.0/ope rator/helper/parser.go:112\\ngithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/parser/syslog.(*Parser).Process\\n\\tgithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.99.0/operator/parser/syslog/parser.go:54\\ngithub.com/ open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper.(*WriterOperator).Write\\n\\tgithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.99.0/operator/helper/writer.go:53\\ngithub.com/open-telemetry/opentelemetry-collector-contrib/p kg/stanza/operator/input/tcp.(*Input).handleMessage\\n\\tgithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.99.0/operator/input/tcp/input.go:191\\ngithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/input/tcp.(*Input).goHan dleMessages.func1\\n\\tgithub.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza@v0.99.0/operator/input/tcp/input.go:152"}
```

### Additional context

This seems to be because we aren't parsing a value into here: https://github.com/influxdata/go-syslog/blob/66067a10754ae90b9540d5312989ae685413c4fe/octetcounting/parser.go#L46 so we get stuck with the default limit

**Comment by sinkingpoint** [comment]
@djaglowski considering that that repo has been archived, would it make sense to fork it here?

**Comment by djaglowski** [comment]
Actually I'm happy to see that the original author has recently created a fork and is making updates again! We should definitely switch in my opinion. https://github.com/leodido/go-syslog.

**Comment by andrzej-stencel** [comment]
PR switching the dependency to the fork: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33205.

**Comment by bacherfl** [comment]
Hi! I would like to pick this issue up if still available

**Comment by bacherfl** [comment]
@djaglowski I went ahead and created a draft PR making use of the new option the updated library. I do have some open questions which I have added to the PR description - appreciate any feedback there

**Comment by djaglowski** [comment]
Resolved by #33777.
